### PR TITLE
[fix](Nereids) DatetimeV2 round floor was incorrectly implemented as round ceil

### PR DESCRIPTION
**Pull request by morrySnow**

### CODE REVIEW

**Comment by morrySnow** [comment]
run buildall

**Comment by morrySnow** [comment]
run buildall

**Comment by morrySnow** [comment]
run buildall

File: fe/fe-core/src/test/java/org/apache/doris/nereids/trees/expressions/literal/DateTimeLiteralTest.java [hunk 1]
```diff
@@ -443,4 +443,36 @@ void testDateTimeV2Scale() {
                 new DateTimeV2Literal(DateTimeV2Type.of(5), "2016-12-31 23:59:59.999999"),
                 new DateTimeV2Literal("2017-01-01 00:00:00.0"));
     }
+
+    @Test
+    void testRoundFloor() {
+        DateTimeV2Literal literal;
+        literal = new DateTimeV2Literal(DateTimeV2Type.of(6), 2000, 2, 2, 2, 2, 2, 222222);
```
Comment (zclllyybb) [inline fe/fe-core/src/test/java/org/apache/doris/nereids/trees/expressions/literal/DateTimeLiteralTest.java:450]: should we also test for input which have more than six decimal places?

Comment (morrySnow) [reply fe/fe-core/src/test/java/org/apache/doris/nereids/trees/expressions/literal/DateTimeLiteralTest.java:450]: no, DateTimeV2Literal has ensure mircosecond could not have more than six scale

**Review by zclllyybb — APPROVED** [review]
LGTM

**Comment by morrySnow** [comment]
run buildall

**Comment by morrySnow** [comment]
run buildall

**Comment by morrySnow** [comment]
run p0

**Review by starocean999 — APPROVED** [review]

**Review by 924060929 — APPROVED** [review]

**Pull request merged**